Moved allocateIterators config to new Collections class

obigu opened the pull request:
Created a new Collections class to hold common configuration/utility methods for all collection classes. Moved allocationIterators from Array class to Collections.

I've made a getter/setter for the property, can change to public as it's not that great anyway.

NathanSweet commented:
I agree putting it on Array is a bit ugly. I wish there was a good place to put it other than to have a whole class for one boolean, but I don't have any better ideas. `isAllocateIterators` quite is an awkward getter name, `getAllocateIterators` would be better (being boolean doesn't require `is`!). However, I think I'd prefer it as just a public field named `allocateIterators`.

obigu commented:
Fixed, replaced by getter/setter by public field. Hopefully there's some common functionality we can add to the new Collections class so that is has a better reason to exist
